### DISCUSSION TITLE
Posts with interview Experiences.

### DISCUSSION THREAD

**Discussion by Nik-git1** [0xku/leetcode-compensation#37] [Ideas]
Can we add some type of marker to posts with interview experience in it(like a star or some small label)? would help someone greatly if they are preparing for a role. We can subsequently add filters for it as well.

**Comment by 0xku** [comment]
Interesting idea. Sounds useful!

Can we do this using some heuristics in a deterministic manner? I don't want to depend on LLMs for this if possible. We are using a relatively smaller LLM, the more we ask it do in the prompt, the chances of output quality dropping goes up. I also don't want to prompt the LLM multiple times for a post (due to it being a free tier model).

It would be great if you can help me scan through the posts manually and try to collect as many patterns as you can that can help us detect if it contains interview experience. Could be as simple as checking for "interview experience" for all you know.

**Reply by Nik-git1** [reply]
Absolutely, we can make a list here itself so others can join in as well.

**Reply by 0xku** [reply]
Great! Folks can use the data/raw_comps.jsonl file to look at the original post to find such patterns. One which looks promising is:
1. "\nInterview Experience": 65 occurrences

**Reply by Nik-git1** [reply]
Cant we just check for a link in the "content", maybe check 
 if "https://leetcode.com/discuss/interview-experience/"substring  is present?

**Reply by Nik-git1** [reply]
I have written a script that might be useful for this , searches the link in post -> stores  2 preceding word and stores in dictionary -> prints post content every time new entry is made into the dictionary for analysis , not printing just the results as they might be insufficient and misleading without the analysis. https://gist.github.com/Nik-git1/36358ba0cee03775a5a02173e588d4be

**Reply by 0xku** [reply]
I don't understand the need to check the previous two words. Are you checking for "interview experience [url]"?

**Reply by Nik-git1** [reply]
Two words right before the url starts (http) so i got results like 1)interview experience 2) interview exparience(typo) 3) interview exp 4) company name(embedded link) , there were 3 4 types of typo variations.

**Reply by 0xku** [reply]
I agree with the idea but let's keep the logic as simple as possible. This pattern looks like is catching all the cases i have seen so far.  You can use https://regex101.com/, paste the content of raw_comps.jsonl there and try this pattern. Make it case insensitive from the options though.
`interview exp(.{22,50})?discuss\/interview-experience`

**Reply by Nik-git1** [reply]
I don't have much experience with LLMs , but can't we just search for the url in the posts? If the url does contain "discuss/interview-experience" we can easily get the link out, we can do without any use of LLMs with a simple function, data is already static.

**Reply by 0xku** [reply]
Ya, I meant that only, no need for LLMs here. I only suggested regex as i thought you are looking for matching "interview exp" and it's variant also along with the url to gain more confidence.

It's fine, you can raise PR that just matches this string in the raw post "https://leetcode.com/discuss/interview-experience".

You can add a star (like this https://www.svgrepo.com/svg/332169/star) to indicate interview experience:
<img width="795" alt="Screenshot 2024-06-23 at 11 14 56 AM" src="https://github.com/kuutsav/leetcode-compensation/assets/79240312/e24c3c3f-4c04-43de-8e79-d5d51b53db5e">

and add a legend below this line "Based on <x> recs parsed between ..." and a filter.

**Reply by Nik-git1** [reply]
Yes sure, I'll get on it.

**Reply by 0xku** [reply]
Thanks
